src/video/SDL_video.c

```diff
@@ -2374,6 +2374,27 @@ SDL_GetWindowMaximumSize(SDL_Window * window, int *max_w, int *max_h)
     }
 }
 
+void
+SDL_GetWindowContentScale(SDL_Window *window, float *h, float *v)
+{
+    CHECK_WINDOW_MAGIC(window,);
+
+    if (_this->GetWindowContentScale) {
+        _this->GetWindowContentScale(_this, window, h, v);
+    } else {
+        int width, height, drawWidth, drawHeight;
+
+        SDL_GetWindowSize(window, &width, &height);
+        SDL_GL_GetDrawableSize(window, &drawWidth, &drawHeight);
```

Comment:
`SDL_GL_GetDrawableSize` is only valid when the window has an OpenGL context associated with it, which might not always be the case (especially if an app is using Vulkan or Metal or SDL_Render, which have their own GetDrawableSize functions in SDL, or when an app hasn't created any rendering surface yet.)

Reply:
Rathar than hacking up things in this new function, shouldn't there be a plain `SDL_GetDrawableSize` that dispatches to the underlying renderer, and maybe falls back to `SDL_GetWindowSize`? (the GL version already does this fallback)

Reply:
> SDL_GetDrawableSize that dispatches to the underlying renderer/ws

This might also get confusing to users if they call it after creating a highdpi-capable window but before attaching any graphics drawable/renderer to it, since it wouldn't return pixel units in that case.

Reply:
I've changed this fallback to use `SDL_GetWindowSizeInPixels` instead, which I hope should be sufficient.